### PR TITLE
Linear filter

### PR DESCRIPTION
**Pull request by maryamabubkr**

### CODE REVIEW

**Comment by reemelbakry** [comment]
Can you write explanatory comments?

File: Task 2/main.m [hunk 1]
```diff
@@ -1 +1,70 @@
...
+   end
+end
+    
+    
+end
+
+figure, imshow(result), title('Filter Applied');
+
+end
+
```
Comment (NayeraMohamed) [inline Task 2/main.m]:
```suggestion
```

File: Task 2/main.m [hunk 2]
```diff
@@ -1 +1,69 @@
...
+    
+ for i = 1:iRows
+   for j= 1:iCols
+      
+           result(i,j)=abs(result(i,j));
+       
+   end
+end
+    
+
```
Comment (NayeraMohamed) [inline Task 2/main.m]:
```suggestion
```

File: Task 2/main.m [hunk 3]
```diff
@@ -1 +1,67 @@
...
+        
+   
+ %absolute...   
+elseif strcmp(Postproc,'absolute')==1
+    
+ for i = 1:iRows
+   for j= 1:iCols
+      
+           result(i,j)=abs(result(i,j));
+
```
Comment (NayeraMohamed) [inline Task 2/main.m]:
```suggestion
```

File: Task 2/main.m [hunk 4]
```diff
@@ -1 +1,66 @@
-%initial commit
\ No newline at end of file
+%initial commit
+function Out = LinearFilter(I, Filter, Postproc) 
+%UNTITLED Summary of this function goes here
+%   Detailed explanation goes here
+
+I= imread("Filters.png");
```
Comment (NayeraMohamed) [inline Task 2/main.m]: Image will be sent as a parameter, we won't read it in the function.

File: Task 2/main.m [hunk 5]
```diff
@@ -1 +1,66 @@
...
+    
+ for i = 1:iRows
+   for j= 1:iCols
+      
+           result(i,j)=abs(result(i,j));
+   end
+end
+end
+
+figure, imshow(result), title('Filter Applied');
```
Comment (NayeraMohamed) [inline Task 2/main.m]:
```suggestion
Out = result;
```

Comment (NayeraMohamed) [reply Task 2/main.m]: or we can directly apply the changes to Out instead of result and return it.

File: Task 2/main.m [hunk 6]
```diff
@@ -1 +1,66 @@
-%initial commit
\ No newline at end of file
+%initial commit
+function Out = LinearFilter(I, Filter, Postproc) 
+%UNTITLED Summary of this function goes here
+%   Detailed explanation goes here
+
+I= imread("Filters.png");
+g = rgb2gray(I);
```
Comment (NayeraMohamed) [inline Task 2/main.m]:
```suggestion
```

Comment (NayeraMohamed) [reply Task 2/main.m]: Lab note: You can apply edge detection for RGB, don’t convert your input to a grayscale.

We can convert it in the main before calling the other masks and remove it from here.

File: Task 2/main.m [hunk 7]
```diff
@@ -1 +1,66 @@
...
+%padding
+[fRows,fCols] = size(Filter);
+rPadding= fix(fRows/2);
+cPadding = fix(fCols/2);
+newR=iRows+(2*rPadding);
+newC= iCols+(2*cPadding);
+temp= zeros(newR,newC );
+temp(rPadding+1:newR-rPadding,cPadding+1:newC-cPadding)=g;
+
+%convolution
```
Comment (NayeraMohamed) [inline Task 2/main.m]: Lab note: Cast to double before apply ANY filter, to keep calculations fractions.

File: Task 2/main.m (hunk 6 above)
Comment (NayeraMohamed) [reply Task 2/main.m]: updated in the main, we can work directly on the I here.

File: Task 2/main.m (hunk 4 above)
Comment (NayeraMohamed) [inline Task 2/main.m]:
```suggestion
```

**Pull request closed without merging**